packages/discussions/src/utils/platform.ts

```diff
@@ -0,0 +1,32 @@
+import { GroupMembership, IGroup, IUser } from "@esri/arcgis-rest-portal";
+
+/**
+ * Utility that returns reducer function that filters a user's groups
+ * by membership type and produces an array of group id's
+ *
+ *
+ * @export
+ * @param {GroupMembership[]} membershipTypes
+ * @return {*}  {((memo: string[], group: IGroup) => string[])}
+ */
+export function reduceByGroupMembership(
+  membershipTypes: GroupMembership[]
+): (memo: string[], group: IGroup) => string[] {
+  return function(memo, group) {
+    if (membershipTypes.indexOf(group.userMembership.memberType) > -1) {
+      memo.push(group.id);
+    }
+    return memo;
+  };
+}
+
+/**
+ * Utility that checks if a user is a portal org admin (default role)
+ *
+ * @export
+ * @param {IUser} user
+ * @return {*}  {boolean}
+ */
+export function isOrgAdmin(user: IUser): boolean {
```

Comment:
That method returns a promise though while my user is already resolved. It would be nice if those rest-js helpers decoupled their argument resolution from the synchronous methods that perform the check/validation

Reply:
ah. would be good to have a comment here to that effect